File: binV9/sixth_problem.rb
require 'json'

staffs = JSON.parse(File.read('input_data/staff.json'), symbolize_names: true)

highestRatedStaffs = staffs.select { |data| data[:result][:rating] > 4 }

mappedResult = highestRatedStaffs.map do |data|
  {
    id: data[:id],
    full_name: data[:full_name],
    email: data[:email],
    rating: data[:result][:rating]
  }
end

puts JSON.pretty_generate({ data: mappedResult })

File: binsV1/binV13/sixth_problem.js
const fs = require("fs");

const staffs = JSON.parse(fs.readFileSync("input_data/staff.json"));

const highestRatedStaffs = staffs.flatMap((staff) => {
  if (staff?.result?.rating <= 4) return [];

  return {
    id: staff.id,
    full_name: staff.full_name,
    email: staff.email,
    rating: staff.result.rating,
  };
});

console.log(JSON.stringify({ data: highestRatedStaffs }, null, 2));
